src/decorator.js

```diff
@@ -19,27 +19,39 @@ class SensorWrapper extends React.Component {
   constructor(props) {
     super(props);
     this.state = {
-      _observables: []
+      _observables: [],
+      sensorFound: false
     };
   }
 
   componentWillMount() {
     const observables = [];
     Object.entries(this.props.sensors).forEach(([name, sensorOptions]) => {
       const options = typeof sensorOptions === "boolean" ? null : sensorOptions;
-      const observable = new Sensors[name](options);
-      observables.push(observable);
 
-      observable.subscribe(sensorValue => {
-        this.setState({
-          [name]: sensorValue
+      const observable = new Sensors[name](options)
```

Comment:
I think this code is a bit confusing, this should also work, right?


```js
new Sensors[name](options).then(observable => {
  observables.push(observable);
  ...
}
```

Maybe we should precompile everything with babel and write something like this 🤔 

```js
try {
  const observable = await new Sensors[name](options);
  observables.push(observable);
} catch(e) { ... }
```

But that is 100% not relevant for this PR 👍

Reply:
Ok I changed it. In order to mantaing functional paradigm I try to avoid try sentence but is just a personal preferring. I also think that if a solution simplifies the code is always welcomed.

Reply:
Also I realise that sensorFound must be an array or object because there are more than one sensor.